services/job_chat/job_chat.py

```diff
@@ -1,96 +1,195 @@
+        history: Optional[List[Dict[str, str]]] = None,
+        context: Optional[str] = None
+    ) -> ChatResponse:
+        """
+        Generate a response using the Claude API with improved error handling and response processing.
+        """
+        history = history.copy() if history else []
+        try:
+            system_message, prompt = build_prompt(content, history, context)
+            
+            message = self.client.beta.prompt_caching.messages.create(
+                max_tokens=self.config.max_tokens,
+                messages=prompt,
+                model=self.config.model,
+                system=system_message
+            )
+
+            # Log cache statistics if available
+            if hasattr(message, 'usage'):
+                if message.usage.cache_creation_input_tokens:
+                    logger.info(f"Cache write: {message.usage.cache_creation_input_tokens} tokens")
+                if message.usage.cache_read_input_tokens:
+                    logger.info(f"Cache read: {message.usage.cache_read_input_tokens} tokens")
+
+            # Process all content types
+            response_parts = []
+            for content_block in message.content:
+                if content_block.type == "text":
+                    response_parts.append(content_block.text)
+                else:
+                    logger.warning(f"Unhandled content type: {content_block.type}")
+
+            response = "\n\n".join(response_parts)
+
+            # Update history with new interaction
+            updated_history = history + [
+                {"role": "user", "content": content},
+                {"role": "assistant", "content": response}
+            ]
+
+            return ChatResponse(
+                content=response,
+                history=updated_history,
+                usage=message.usage.model_dump() if hasattr(message, 'usage') else {}
+            )
+
+        except Exception as e:
+            logger.error(f"Error during generation: {str(e)}", exc_info=True)
+            raise
+
+def main(data_dict: dict) -> dict:
+    """
+    Main entry point with improved error handling and input validation.
+    """
     try:
-        logger.info("Generating")
-
-        message = client.beta.prompt_caching.messages.create(
-            max_tokens=max_tokens, messages=prompt, model=claude_model, system=system_message
+        data = Payload.from_dict(data_dict)
+        
+        # Create client with custom config if API key provided
+        config = ChatConfig(api_key=data.api_key) if data.api_key else None
+        client = ClaudeClient(config)
+        
+        result = client.generate(
+            content=data.content,
+            history=data_dict.get("history", []),
+            context=data.context
+        )
+        
+        return {
+            "response": result.content,
+            "history": result.history,
+            "usage": result.usage
+        }
+        
+    except APIConnectionError as e:
```

Comment:
This is fantastic :exclamation: 

I was worried that mapping each error was a bit of a pain, but actually this gives us wonderfully clean and comprehensive error messages. Nice work!